File: SecureBlackbox Samples/Office Encryptor/java/officeencryptor.java
/*
 * SecureBlackbox 2022 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of SecureBlackbox in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/secureblackbox
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import secureblackbox.*;

public class officeencryptor extends ConsoleDemo {

	public static Certificate loadCertificate(String file, String password)
	{
		Certificate cert = null;

		if (file.length() > 0)
		{
			try
			{
				Certificatemanager certmanager = new Certificatemanager();

				certmanager.importFromFile(file, password);

				cert = certmanager.getCertificate();
			}
			catch (Exception e)
			{
				System.out.println("Cannot load certificate!");
			}
		}

		return cert;
	}

	public static void main(String[] args) {
		if (args.length < 6) {
			System.out.println("usage: officeencryptor [-input input_file] [-output output_file] [-pass encryption_password] [-enctype encryption_type] [-encalg encryption_algorithm]");
			System.out.println("Options: ");
			System.out.println("  -input        An input file to encrypt (Required).");
			System.out.println("  -output       Where the encrypted file will be saved (Required).");
			System.out.println("  -pass         Password for file encryption (Required).");
			System.out.println("  -enctype      The type of encryption to use. Enter the corresponding number. Valid values:");
			System.out.println("                  0 - oetDefault");
			System.out.println("                  1 - oetBinaryRC4");
			System.out.println("                  2 - oetBinaryRC4CryptoAPI");
			System.out.println("                  3 - oetOpenXMLStandard");
			System.out.println("                  4 - oetOpenXMLAgile");
			System.out.println("                  5 - oetOpenDocument");
			System.out.println("  -encalg       The encryption algorithm to use. Valid values: RC2, RC4, DES, 3DES, AES128, AES192, AES256, Blowfish");
			System.out.println("\r\nExample: officeencryptor -input C:\\xml\\myfile.xml -output C:\\xml\\mysignedfile.xml -cert C:\\certs\\mycert.pfx -certpass mypassword");
		} else {
			try {
				Officeencryptor encryptor = new Officeencryptor();
				System.out.println("***************************************************************************************************");
				System.out.println("* This sample illustrates the use of OfficeEncryptor component for encrypt office documents. *");
				System.out.println("***************************************************************************************************\n");

				for (int i = 0; i < args.length; i++) {
					if (args[i].startsWith("-")) {
						if (args[i].equals("-input"))
							encryptor.setInputFile(args[i + 1]); // args[i+1] corresponds to the value of argument [i]
						if (args[i].equals("-output"))
							encryptor.setOutputFile(args[i + 1]);
						if (args[i].equals("-pass"))
							encryptor.setPassword(args[i + 1]);
						if (args[i].equals("-enctype"))
							encryptor.setEncryptionType(Integer.parseInt(args[i + 1]));
						if (args[i].equals("-encalg"))
							encryptor.setEncryptionAlgorithm(args[i + 1]);
					}
				}

				encryptor.encrypt();

				System.out.println("Office document successfully encrypted");
			}
			catch (Exception ex)
			{
				displayError(ex);
			}
		}
	}
}


class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }

  static String prompt(String label, String punctuation, String defaultVal)
  {
	System.out.print(label + " [" + defaultVal + "] " + punctuation + " ");
	String response = input();
	if(response.equals(""))
		return defaultVal;
	else
		return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof SecureBlackboxException) {
      System.out.print(" (" + ((SecureBlackboxException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }
}

File: SecureBlackbox Samples/Office Encryptor/php/office_encryptor.php
<?php
/*
 * SecureBlackbox 2024 PHP Edition - Sample Project
 *
 * This sample project demonstrates the usage of SecureBlackbox in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/secureblackbox
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
require_once('../include/secureblackbox_officeencryptor.php');
require_once('../include/secureblackbox_certificatemanager.php');
require_once('../include/secureblackbox_const.php');
?>
<style>
table { width: 100% !important; }
td { white-space: nowrap; }
td input { width: 100%; }
td:last-child { width: 100%; }
</style>

<div width="90%">
  <form method=POST>
    <h2>Office Encryptor Demo</h2>
    
    <h3>General Options</h3>
    <table>
      <tr><td>Input File:</td><td><input type=text name=inputFile value=""></td></tr>
      <tr><td>Output File:</td><td><input type=text name=outputFile value=""></td></tr>
    </table>
    <br/>
    <br/>

    <b>Encryption Options</b><br/>
    <table>
      <tr>
        <td>Encryption Type:</td>
        <td>
          <select name="encType">
            <option value="OFFICEENCRYPTOR_ENCRYPTIONTYPE_DEFAULT">Default</option>
            <option value="OFFICEENCRYPTOR_ENCRYPTIONTYPE_BINARY_RC4">BinaryRC4</option>
            <option value="OFFICEENCRYPTOR_ENCRYPTIONTYPE_BINARY_RC4CRYPTO_API">BinaryRC4CryptoAPI</option>
            <option value="OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_XMLSTANDARD">OpenXMLStandard</option>
            <option value="OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_XMLAGILE">OpenXMLAgile</option>
            <option value="OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_DOCUMENT">OpenOffice</option>
          </select>
        </td>
      </tr>
      <tr>
        <td>Encryption Algorithm:</td>
        <td>
          <select name="encAlg">
            <option value=""></option>
            <option value="RC2">RC2</option>
            <option value="RC4">RC4</option>
            <option value="DES">DES</option>
            <option value="3DES">3DES</option>
            <option value="AES128">AES128</option>
            <option value="AES192">AES192</option>
            <option value="AES256">AES256</option>
            <option value="Blowfish">Blowfish</option>
          </select>
        </td>
      </tr>
      <tr><td>Password:</td><td><input type=password name=encpass value=""></td></tr>
    </table>
    <br/>
    <br/>

    <input type="submit" value="Encrypt" />
  </form>
</div><br/>

<?php

function translateEncryptionType($encType){
  switch($encType){
    case "OFFICEENCRYPTOR_ENCRYPTIONTYPE_BINARY_RC4":  return OFFICEENCRYPTOR_ENCRYPTIONTYPE_BINARY_RC4; break;
    case "OFFICEENCRYPTOR_ENCRYPTIONTYPE_BINARY_RC4CRYPTO_API":  return OFFICEENCRYPTOR_ENCRYPTIONTYPE_BINARY_RC4CRYPTO_API; break;
    case "OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_XMLSTANDARD":  return OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_XMLSTANDARD; break;
    case "OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_XMLAGILE":  return OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_XMLAGILE; break;
    case "OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_DOCUMENT":  return OFFICEENCRYPTOR_ENCRYPTIONTYPE_OPEN_DOCUMENT; break;
    default: return OFFICEENCRYPTOR_ENCRYPTIONTYPE_DEFAULT; break;
  }
}

  if ($_SERVER['REQUEST_METHOD'] == "POST") {
    $officeencryptor = new SecureBlackbox_OfficeEncryptor();
    
    try {
      // General options
      $officeencryptor->setInputFile($_REQUEST['inputFile']);
      $officeencryptor->setOutputFile($_REQUEST['outputFile']);
      
      // Encryption options
      $officeencryptor->setEncryptionType(translateEncryptionType($_REQUEST['encType']));
      $encAlg = $_REQUEST['encAlg'];
      if (!empty($encAlg)) {$officeencryptor->setEncryptionAlgorithm($encAlg);}
      $officeencryptor->setPassword($_REQUEST['encpass']);

      $officeencryptor->doEncrypt();
      echo "<h2>Office document successfully encrypted</h2>";
    }
    catch (exception $e) {
      echo "<h2>Encryption Failure (Details Below)</h2><p>" . $e->getMessage() . "</p>";
    }
  }
?>